examples/chip-tool/commands/clusters/Commands.h

```diff
@@ -21,6 +21,369 @@
 #pragma once
 
 #include "../common/ModelCommand.h"
+#include "../common/ModelCommandResponse.h"
+
+#include <support/SafeInt.h>
+
+bool ReadAnyType(uint8_t *& message)
+{
+    uint8_t type = chip::Encoding::Read8(message);
+    ChipLogProgress(chipTool, "  type: 0x%02x", type);
+
+    // FIXME: Should we have a mapping of type ids to types, based on
+    // table 2.6.2.2 in Rev 8 of the ZCL spec?
+    switch (type)
+    {
+    case 0x00: // nodata / No data
+        ChipLogError(chipTool, "Type 0x%02x is not supported", type);
+        message += 0;
+        break;
+    case 0x08: // data8 / 8-bit data
+        ChipLogProgress(chipTool, "  value: 0x%02x", chip::Encoding::Read8(message));
```

Comment:
Here and in all the other branches: how do we know this is not reading off the end of the buffer?